bam-to-unmapped-bams.wdl

```diff
@@ -0,0 +1,164 @@
+## Copyright Broad Institute, 2017
+## 
+## This WDL converts BAM  to unmapped BAMs
+##
+## Requirements/expectations :
+## - BAM file
+##
+## Outputs :
+## - Sorted Unmapped BAMs
+##
+## Cromwell version support
+## - Successfully tested on v31
+## - Does not work on versions < v23 due to output syntax
+##
+## Runtime parameters are optimized for Broad's Google Cloud Platform implementation. 
+## For program versions, see docker containers. 
+##
+## LICENSING : 
+## This script is released under the WDL source code license (BSD-3) (see LICENSE in 
+## https://github.com/broadinstitute/wdl). Note however that the programs it calls may 
+## be subject to different licenses. Users are responsible for checking that they are
+## authorized to run all programs before running this script. Please see the docker 
+## page at https://hub.docker.com/r/broadinstitute/genomes-in-the-cloud/ for detailed
+## licensing information pertaining to the included programs.
+
+# WORKFLOW DEFINITION
+workflow BamToUnmappedBams {
+  File input_bam
+
+  Int? additional_disk_size
+  Int additional_disk = select_first([additional_disk_size, 20])
+
+  Float input_size = size(input_bam, "GB")
+  
+  String? gatk_path
+  String path2gatk = select_first([gatk_path, "/gatk/gatk"])
+
+  String? gitc_docker
+  String gitc_image = select_first([gitc_docker, "broadinstitute/genomes-in-the-cloud:2.3.1-1512499786"])
+  String? gatk_docker 
+  String gatk_image = select_first([gatk_docker, "broadinstitute/gatk:latest"])
+
+  call GenerateOutputMap {
+    input:
+      input_bam = input_bam,
+      disk_size = ceil(input_size) + additional_disk,
+      docker = gitc_image
+  }
+
+  call RevertSam {
+    input:
+      input_bam = input_bam,
+      output_map = GenerateOutputMap.output_map,
+      disk_size = ceil(input_size * 3) + additional_disk,
+      docker = gatk_image,
+      gatk_path = path2gatk
+  }
+
+  scatter (unmapped_bam in RevertSam.unmapped_bams) {
+    String output_basename = basename(unmapped_bam, ".coord.sorted.unmapped.bam")
+    Float unmapped_bam_size = size(unmapped_bam, "GB")
+
+    call SortSam {
+      input:
+        input_bam = unmapped_bam,
+        sorted_bam_name = output_basename + ".unmapped.bam",
```

Comment:
shouldn't need the `+ ".unmapped.bam"` anymore (as `output_basename` includes that extension)

Reply:
I think `basename(unmapped_bam, ".coord.sorted.unmapped.bam")` removes the `unmapped.bam` part so this line adds back without the `.coord.sorted`

Reply:
fudge--YES